interfaces/builtin/cifs_mount_control.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const cifsMountControlSummary = `allows mounting and unmounting CIFS filesystems`
+
+const cifsMountControlBaseDeclarationSlots = `
+  cifs-mount-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const cifsMountControlConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access.
+
+mount
+umount
+umount2
+`
+
+const cifsMountControlConnectedPlugAppArmor = `
+# Description: Allow mounting and unmounting CIFS filesystems.
+
+# Required for mounts and unmounts
+capability sys_admin,
+
+# Allow mounts to our snap-specific writable directories
+mount fstype=cifs ** -> @{HOMEDIRS}/*/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+mount fstype=cifs ** -> @{HOMEDIRS}/*/snap/@{SNAP_NAME}/common/{,**},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/common/{,**},
```

Comment:
Using '**' means that the snap could try to mount anything from the host into SNAP_DATA or SNAP_COMMON, which would bypass intended access restrictions. That said, the fstype should cause things to fail, but a crafted mount command shipped in the snap might be used to tickle kernel bugs. What do denials look like when you try to mount cifs filesystems and you don't have these rules?

Reply:
```
= AppArmor =
Time: Jul 15 05:52:53
Log: apparmor="DENIED" operation="mount" info="failed mntpnt match" error=-13 profile="snap.smb-copier.smb-copier" name="/var/snap/smb-copier/common/shares/" pid=14381 comm="mount.cifs" fstype="cifs" srcname="//192.168.123.1/sambashare"
```

Reply:
@jdstrand what would you suggest otherwise? I would like to allow all kind of cifs options and arbitrary cifs pathes. Would it help to let the mount pathes start with // ? How would I allow all mount options and pathes starting with `//` ?

Reply:
It would look better if we could do:
```
mount fstype=cifs //** -> /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
mount fstype=cifs //** -> /var/snap/@{SNAP_NAME}/common/{,**},
```

Can you try it locally and see if it works? I suspect however that it will not and that apparmor_parser will minimize that to '/**' internally and it won't match. If it doesn't work, simply use:

```
// source is of form '//<ip>/dir1/dirN' and is therefore not predictable so we'll
// rely on the fstype.
mount fstype=cifs -> /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
mount fstype=cifs -> /var/snap/@{SNAP_NAME}/common/{,**},
```